Generation/src/components/GaussSmearVertex.cpp

```diff
@@ -0,0 +1,79 @@
+#include "GaussSmearVertex.h"
+
+#include "GaudiKernel/DeclareFactoryEntries.h"
+#include "GaudiKernel/IRndmGenSvc.h"
+#include "GaudiKernel/PhysicalConstants.h"
+#include "GaudiKernel/SystemOfUnits.h"
+#include "GaudiKernel/Vector4DTypes.h"
+
+/// Declaration of the Tool Factory
+DECLARE_TOOL_FACTORY(GaussSmearVertex)
+
+/// Standard constructor, initializes variables
+GaussSmearVertex::GaussSmearVertex(const std::string& type, const std::string& name, const IInterface* parent)
+    : GaudiTool(type, name, parent) {
+  declareInterface<IVertexSmearingTool>(this);
+
+}
+
+/// Destructor
+GaussSmearVertex::~GaussSmearVertex() { ; }
+
+//=============================================================================
+// Initialize
+//=============================================================================
+StatusCode GaussSmearVertex::initialize() {
+  StatusCode sc = GaudiTool::initialize();
+  if (sc.isFailure()) return sc;
+
+  IRndmGenSvc* randSvc = svc<IRndmGenSvc>("RndmGenSvc", true);
+
+
+
+  sc = m_gaussDist.initialize(randSvc, Rndm::Gauss(0., 1));
+  if (sc.isFailure()) return sc;
+
+  std::string infoMsg = " applying TOF of interaction with ";
+  if (m_zDir == -1) {
```

Comment:
I don't see where this is used in the code except for these messages? Is the member necessary?